create_shards.c

```diff
@@ -91,12 +103,69 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("could not find column: %s", partitionColumnName)));
 	}
 
-	/* we only support hash partitioning method for now */
-	if (partitionMethod != HASH_PARTITION_TYPE)
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+	partitionColumnTypeId = partitionColumnVar->vartype;
+
+	if (partitionMethod == HASH_PARTITION_TYPE)
+	{
+		partitionColumnOpClassId = GetDefaultOpClass(partitionColumnTypeId, HASH_AM_OID);
+	}
+	else
 	{
+		/* we only support hash partitioning right now */
 		ereport(ERROR, (errmsg("unsupported partition method: %c", partitionMethod)));
 	}
 
+	/*
+	 * If data type of the partition column does not have a default operator class, we
+	 * should prevent distribution of the table on this column. It is because there are
+	 * no operators/procedures that can manipulate the value of the column.
+	 */
+	if (partitionColumnOpClassId == InvalidOid)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
```

Comment:
`ERRCODE_WRONG_OBJECT_TYPE` isn't appropriate here. Let's stick with what other functions do when they get an `InvalidOid` here. I think the following should work:

```c
ereport(ERROR,
		(errcode(ERRCODE_UNDEFINED_OBJECT),
		 errmsg("data type %s has no default operator class for specified partition method",
		 		format_type_be(attrType)),
		 errdetail("Partition column types must have a default operator class defined.")));
```